### ISSUE TITLE
Linker error when building google_iot_mqtt sample with zephyr-sdk 0.11.1

### ISSUE THREAD

**Issue by tsvehagen** [zephyrproject-rtos/zephyr#22484]
**Describe the bug**
After upgrading to zephyr-sdk 0.11.1 I got a linker error when building `samples/net/cloud/google_iot_mqtt`. The linker complains about `_gettimeofday` and seems to be the same error as in https://github.com/zephyrproject-rtos/zephyr/issues/13384. 

**To Reproduce**
1. west build -b nrf52840_pca10056 -d build/iot-sample zephyr/samples/net/cloud/google_iot_mqtt

**Screenshots or console output**
```
/opt/zephyr-sdk/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/9.2.0/../../../../arm-zephyr-eabi/bin/ld: /opt/zephyr-sdk/arm-zephyr-eabi/arm-zephyr-eabi/lib/thumb/v7e-m/nofp/libc_nano.a(lib_a-gettimeofdayr.o): in function `_gettimeofday_r':
gettimeofdayr.c:(.text._gettimeofday_r+0xe): undefined reference to `_gettimeofday'
collect2: error: ld returned 1 exit status
```

**Environment (please complete the following information):**
 - OS: ArchLinux
 - Toolchain: zephyr-sdk 0.11.1
 - SHA: f21aaeffaa5ed10629915a9d198ff483865f7be9

**Comment by mike-scott** [comment]
To add to this, I don't see this error using the 0.10.3 SDK with basically the same setup:

- OS: ArchLinux
- Toolchain: zephyr-sdk 0.10.3
- SHA: 4fea711f (today's master)

**Comment by carlescufi** [comment]
Set as high priority since this is a regression from SDK 0.10.x

**Comment by SebastianBoe** [comment]
reproduced

**Comment by SebastianBoe** [comment]
I'm observing that it affects 0.11.1, but not 0.11.0.

@galak : Are you aware of any changes made between 0.11.0 and 0.11.1 that could cause this?

**Comment by SebastianBoe** [comment]
Strange, this is the only commit I can find in 0.11.1, and it presumably can't cause this.

https://github.com/zephyrproject-rtos/sdk-ng/commit/df84499eba3fb1c97592506a19d060113b64483d

**Comment by SebastianBoe** [comment]
Ah, it could be related to the build scripts that change behaviour depending on the SDK version.

**Comment by SebastianBoe** [comment]
e.g. perhaps nano is used in 0.11.1, but not 0.11.0, and perhaps nano doesn't have _gettimeofday.

**Comment by SebastianBoe** [comment]
Okay, I have the answer now.

zephyr-sdk-0.11.1/arm-zephyr-eabi/arm-zephyr-eabi/lib/thumb/v7e-m/nofp/libc_nano.a

has been built incorrectly.

This must be fixed by the user that built zephyr-sdk-0.11.1/arm-zephyr-eabi
@galak ?

This is a good build:

Good:

```
lib_a-gettimeofdayr.o:
         U errno
         U gettimeofday
00000000 T _gettimeofday_r

lib_a-sysgettod.o:
00000000 T gettimeofday
         U _gettimeofday_r
         U _impure_ptr

lib_a-time.o:
         U _gettimeofday_r
         U _impure_ptr
00000000 T time

```
This is a bad build:

```
lib_a-gettimeofdayr.o:
         U errno
         U _gettimeofday
00000000 T _gettimeofday_r

lib_a-sysgettod.o:
00000000 T gettimeofday
         U _gettimeofday_r
         U _impure_ptr

lib_a-time.o:
         U _gettimeofday_r
         U _impure_ptr
00000000 T time


```

Note that "lib_a-gettimeofdayr.o" should be referencing 'gettimeofday', not '_gettimeofday'.

**Comment by SebastianBoe** [comment]
The good build is from 0.11.0. The bad build is from 0.11.1.
